src/runtime/methodbinder.cs

```diff
@@ -342,20 +342,53 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                 bool paramsArray;
                 int kwargsMatched;
                 int defaultsNeeded;
-
-                if (!MatchesArgumentCount(pynargs, pi, kwargDict, out paramsArray, out defaultArgList, out kwargsMatched, out defaultsNeeded))
+                bool isOperator = OperatorMethod.IsOperatorMethod(mi);
+                if (!MatchesArgumentCount(pynargs, pi, kwargDict, isOperator, out paramsArray, out defaultArgList, out kwargsMatched, out defaultsNeeded))
                 {
                     continue;
                 }
                 var outs = 0;
+                int clrnargs = pi.Length;
+                isOperator = isOperator && pynargs == clrnargs - 1;  // Handle mismatched arg numbers due to Python operator being bound.
```

Comment:
If pynargs == clrnargs - 1, then we know we have the correct number of arguments.  So I think the logic can be simplified by including (pynargs == clrnargs - 1) in the original definition of isOperator, not passing isOperator to MatchesArgumentCount, and using 
```csharp
if (!MatchesArgumentCount(...) && !isOperator)
```

Reply:
Made this change but reverted it since it breaks other test cases involving default args, see: https://github.com/pythonnet/pythonnet/pull/1324/checks?sha=8cce61dd5def72485e599a071e9fd3bfc12346c6

Reply:
You didn't implement the change correctly.  See below.